modules/ethereum/src/lib.rs

```diff
@@ -112,28 +155,13 @@ pub struct ImportContext<Submitter> {
 	submitter: Option<Submitter>,
 	parent_header: Header,
 	parent_total_difficulty: U256,
-	next_validators_set_id: u64,
-	next_validators_set: (H256, Vec<Address>),
+	parent_scheduled_change: Option<ScheduledChange>,
+	validators_set_id: u64,
+	validators_set: ValidatorsSet,
+	last_signal_block: Option<H256>,
 }
 
 impl<Submitter> ImportContext<Submitter> {
```

Comment:
Github won't let me comment on the struct since it's not part of the PR, but I'm not sure what the `ImportContext` represents. Can you clarify that for me?

Reply:
There are 2 groups of header verification rules. Rules from first group can be checked just by looking at the header itself, without knowing its parent and blockchain state (not the regular blockchain state - just something that is associated with parent header). 2nd group requires this context. `ImportContext` meant to store everything we need for checking 2nd group rules.

Reply:
Okay, so to be clear the context is  extra information needed to validate that the current header is valid. In this case the context needed is information associated with the current header's parent, yes?

I know it's not part of the PR, but I think having a more descriptive comment as part of the `ImportContext` struct would be helpful :)

Reply:
Looks true :)

Could you suggest documentation that you'd like to see? I do not want something like "it is extra information for..." or "it has everything required to...". But I can't suggest something other than "It is used to verify header within blockchain context", which I believe isn't something you wanted to see there :)

Reply:
Sure. What do you think of this?

```
/// Header import context.
///
/// The import context contains information needed by the header verification
/// pipeline which is not directly part of the header being imported. This includes
/// information relating to its parent, and the current validator set (which
/// provide _context_ for the current header).
```